LEGO1/mxdirectdraw.cpp

```diff
@@ -1,22 +1,1223 @@
+
 #include "mxdirectdraw.h"
+#include "decomp.h"
 
-// OFFSET: LEGO1 0x1009e7f0
-int MxDirectDraw::FlipToGDISurface()
+
+#ifndef DDSCAPS_3DDEVICE
+#define DDSCAPS_3DDEVICE 0x00002000l
+#endif
+
+// GLOBAL OFFSET: LEGO1 0x10100C68
+extern BOOL g_is_PALETTEINDEXED8 = 0;
+
+// OFFSET: LEGO1 0x1009DA20
+void EnableResizing(HWND hwnd, BOOL flag)
+{
+  static DWORD dwStyle;
+
+  if (!flag)
+  {
+    dwStyle = GetWindowLong(hwnd, GWL_STYLE);
+    if (dwStyle & WS_THICKFRAME)
+    {
+      SetWindowLong(hwnd, GWL_STYLE, GetWindowLong(hwnd, GWL_STYLE) ^ WS_THICKFRAME);
+    }
+  }
+  else
+  {
+    SetWindowLong(hwnd, GWL_STYLE, dwStyle);
+  }
+}
+
+// OFFSET: LEGO1 0x1009EFD0
+MxDirectDraw::DeviceModesInfo::~DeviceModesInfo()
+{
+  if (p_guid != NULL)
+  {
+    free(p_guid);
+  }
+
+  if (m_mode_ARRAY != NULL)
+  {
+    free(m_mode_ARRAY);
+  }
```

Comment:
I agree these are probably more likely `delete`, which will end up inlining the `NULL` check. 
```suggestion
  delete p_guid;
  delete m_mode_ARRAY;
```